### ISSUE TITLE
training parameters for cityscapes 

### ISSUE THREAD

**Issue by Engro39** [TexasInstruments/edgeai-modelzoo#4]
I am working on the training for semantic segmentation with cityscapes dataset. The model I want to train is 'RegNetX800MF+FPNLite' Its mIOU is 72.01.

Could you share the details of training parameters to reproduce the mIOU? 

For instance
 - lr
 - epochs
 - optimizer
 - scheduler
 - ...

Also, did you use 'valid' dataset to get the mIOUs on the table?

**Comment by mathmanu** [comment]
This may be what you are using for: https://github.com/TexasInstruments/edgeai-torchvision/blob/master/run_edgeailite_segmentation.sh#L84

We used the gtFine set and 19 classes that are supposd to be used for training and valuation.

**Comment by Engro39** [comment]
thank you @mathmanu. Do you know what dataset was used for the mIOU on the table?

![image](https://user-images.githubusercontent.com/13134640/184634737-fd409a94-256a-4d6c-9354-d5bdfee301c5.png)

**Comment by mathmanu** [comment]
It says Cityscapes in the table. What is your doubt?

It was generated by this training script: https://github.com/TexasInstruments/edgeai-torchvision/blob/master/run_edgeailite_segmentation.sh#L84

**Comment by Engro39** [comment]
@mathmanu , I mean there are two sets in cityscapes dataset, 'train' and 'valid'. My question is what set among the 'train' and 'valid' was used to get the number?

**Comment by Engro39** [comment]
Also, in the most cases, the mIOU in valid set is lower than in train set. just want to make sure what set is used for to reproduce the performance.

**Comment by mathmanu** [comment]
The accuracies reported are on the validation ('val') split.

**Comment by Engro39** [comment]
Thank you for the confirmation.

**Comment by Engro39** [comment]
@mathmanu , I have done with the training for "fpn_aspp_regnetx800mf_edgeailite" which is "RegNetX800MF + FPNLite" and the mIOU is 67.67% for the validation split. I used the same configuration as you mentioned. However, on the table, the mIOU is 72.01. There is about 4% difference. Is there anything I was missing?

**Comment by mathmanu** [comment]
I have attached my full training log here. You can at epoch 207 the MeanIoU reported is 72.01. That is the model that I have selected.
[fpn_aspp_regnetx800mf_edgeailite_768x384_20200911.log](https://github.com/TexasInstruments/edgeai-modelzoo/files/9399709/fpn_aspp_regnetx800mf_edgeailite_768x384_20200911.log)

Your training accuracy can be lower if your pretrained weights loading did not happen correctly if any of the hyper parameters are not same.

**Comment by Engro39** [comment]
@mathmanu , I compared all the configurations in the logs. I found three.

![image](https://user-images.githubusercontent.com/13134640/186762679-5b8b9795-9a35-4afb-bce0-f313fd41ea33.png)

Do you think these could affect the degradation? Also, I couldn't find the model of 'fpnlite_pixel2pixel_aspp_regnetx800mf' in the git repo.

**Comment by mathmanu** [comment]
It's the same model, the name has changed recently. I don't think the changes that you highlighted will affect accuracy.

If for some reason, the pretrained model loading is not happening correctly, it can affect the accuracy. 
Also if the number of GPUs used for training is large, it can affect accuracy since I am not using SyncBatchNorm and the batchsize used are small - I think I might have used 2 GPU. 
Also check if the learning rate matches with what I used.

**Comment by Engro39** [comment]
@mathmanu , One thing I found is I did set the batch size as 64, so I changed it to 16 and trained the model again. The final result was 71.09 at 197 epoch. There is about 1% difference, but I think it's acceptable. The batch size was the reason I guess.

**Comment by mathmanu** [comment]
Try with batch size of 12 and 2 gpus - I think that is what I used.

**Comment by Engro39** [comment]
In your log, I see the batch size is 16, but let me try with 12 though.

**Comment by Engro39** [comment]
@mathmanu 72.24 @ 198 epoch with 12 batch size. Thank you for your comments.
